Microsoft.Identity.Web/Client/TokenCacheProviders/InMemory/MSALAppMemoryTokenCacheProvider.cs

```diff
@@ -0,0 +1,147 @@
+﻿/************************************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2015 Microsoft Corporation
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+***********************************************************************************************/
+
+using Microsoft.AspNetCore.Authentication.AzureAD.UI;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Identity.Client;
+using System;
+
+namespace Microsoft.Identity.Web.Client.TokenCacheProviders
+{
+    /// <summary>
+    /// An implementation of token cache for Confidential clients backed by MemoryCache.
+    /// MemoryCache is useful in Api scenarios where there is no HttpContext to cache data.
+    /// </summary>
+    /// <seealso cref="https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/token-cache-serialization"/>
+    public class MSALAppMemoryTokenCacheProvider : IMSALAppTokenCacheProvider
+    {
+        /// <summary>
+        /// The application cache key
+        /// </summary>
+        internal string AppCacheId;
+
+        /// <summary>
+        /// The backing MemoryCache instance
+        /// </summary>
+        internal IMemoryCache memoryCache;
+
+        /// <summary>
+        /// The duration till the tokens are kept in memory cache. In production, a higher value , upto 90 days is recommended.
+        /// </summary>
+        private readonly DateTimeOffset cacheDuration = DateTimeOffset.Now.AddHours(12);
```

Comment:
@kalyankrishna1 : why do we need it? is it a best practice?
should it be in the configuration if the library was reusable?

Reply:
Removed and replaced with an Options class

---
In reply to: [267388158](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/65#discussion_r267388158) [](ancestors = 267388158)